linode/manager
# test: [M3-6876] - Improve formatting of GitHub test result comments

## Description 📝
This PR makes some slight improvements to our GitHub comment test result formatter script. It improves the formatting of the breakdown showing the number of passing, failing, and skipped tests, adds a customizable title, and changes the "Debugging" heading to "Troubleshooting".

Note that this is a low priority and our separate changes to enable GitHub Comments in CI do not rely on these changes being in place first.


## Changes  🔄
- Add pipeline title to GitHub comment
- Improve heading formatting
- Render test result summary as a table
- Change "Debugging" to "Troubleshooting"

## Target release date 🗓️
N/A, this is not a high priority at all.

## Preview 📷

**Passing:**
![Screenshot 2024-10-31 at 3 19 06 PM](https://github.com/user-attachments/assets/934ec72e-1d6c-438b-a9ee-bbe4910223e6)

**With Failures:**
![Screenshot 2024-10-31 at 3 25 13 PM](https://github.com/user-attachments/assets/43202e8e-cc42-4c1d-a142-8a7af7593c40)

I didn't take any screenshots showing the before state, but the table showing the number of passing, failing, and skipped tests was previously a one liner along the lines of " :x: N failing tests | 💚 N passing tests | 🔽 N skipped tests", and the "Cloud Manager UI Test Results" heading was not present.

## How to test 🧪

You can manually run the `yarn junit:summary <path> --format github` command and validate its output, or check out the comments posted to my test PR on my fork: https://github.com/jdamore-linode/manager/pull/25

Soon I'll add some links to Jenkins run to the Jira ticket showing how it all fits together.

## As an Author I have considered 🤔

*Check all that apply*
